assembly/src/parsers/mod.rs

```diff
@@ -121,6 +135,14 @@ fn parse_op_token(
         _ => return Err(AssemblyError::invalid_op(op)),
     }?;
 
+    if in_debug_mode {
+        let dec_idx = decorators[dec_len].0;
```

Comment:
nit: is `dec_idx` the right name here? I think this value represents the start of the operation - so, maybe `op_start` or  `asmop_start` would be a more appropriate name?